common/unified/matrix/dense_kernels.template.cpp

```diff
@@ -456,6 +456,33 @@ void advanced_row_gather(std::shared_ptr<const DefaultExecutor> exec,
 }
 
 
+template <typename ValueType, typename OutputType, typename IndexType>
+void row_scatter(std::shared_ptr<const DefaultExecutor> exec,
+                 const array<IndexType>* row_idxs,
+                 const matrix::Dense<ValueType>* orig,
+                 matrix::Dense<OutputType>* target, bool& invalid_access)
+{
+    array<bool> invalid_access_arr{exec, {false}};
+    run_kernel(
+        exec,
+        [num_rows = target->get_size()[0]] GKO_KERNEL(
+            auto row, auto col, auto orig, auto rows, auto scattered,
+            auto* invalid_access_ptr) {
+            if (rows[row] >= num_rows) {
+                *invalid_access_ptr = true;
+                return;
+            }
```

Comment:
maybe leave a TODO hear for when we have established a uniform error handling framework.